browser/ntp_background/ntp_p3a_helper_impl.cc

```diff
@@ -214,16 +272,71 @@ void NTPP3AHelperImpl::OnP3AMetricCycled(const std::string& histogram_name,
                                      creative_instance_id);
 }
 
-std::string NTPP3AHelperImpl::BuildHistogramName(
-    const std::string& creative_instance_id,
-    const std::string& event_type) {
-  return base::StrCat(
-      {p3a::kCreativeMetricPrefix, creative_instance_id, ".", event_type});
+void NTPP3AHelperImpl::CleanOldCampaignsAndCreatives() {
+  ScopedDictPrefUpdate update(local_state_, kNewTabPageKnownCampaignsDictPref);
+
+  if (!p3a_service_->IsP3AEnabled()) {
+    update->clear();
+    return;
+  }
+
+  const auto now = base::Time::Now();
+
+  auto check_expiry = [&](const base::Value::Dict* dict) -> bool {
```

Comment:
This lambda is doesn't require external context. And seems like it can be moved to a free function in an anonymous namespace.